src/Chirp.Infrastructure/Repository/AuthorRepository.cs

```diff
@@ -122,6 +123,30 @@ public ICollection<Cheep> GetCheepsByAuthor(Guid id, int page)
         if(author.Cheeps.Count > 32) return author.Cheeps.ToList<Cheep>().GetRange(pageSizeIndex,PageSize);
         return author.Cheeps;
     }
+    
+    public ICollection<Cheep> GetCheepsByAuthorAndFollowing(Guid id, int page)
+    {
+        Author author = GetAuthorById(id);
+        ICollection<Cheep> cheeps = new List<Cheep>(author.Cheeps); 
+
+        foreach (Author follower in author.Following)
+        {
+            ICollection<Cheep> followerCheeps = GetCheepsByAuthor(follower.Id, page);
```

Comment:
Getting cheeps by page for every author including the user itself, won't return all cheeps.
Pose a scenario where some authors only have cheeps to fill page 1 and we are trying to get cheeps from page two. 

To resolve the problem, we need to gather every cheep from every relevant author and then perform the pagination!

Reply:
Too add to this.
Methods for gathering every cheep from an author without paging is the way forward

Reply:
That is a very good observation. I will make sure to make a fix.

Reply:
The fix for this will be included in another PR: The current PR will focus on fixing the View logic, that determines which cheeps to show on a timeline, according to the users login status.

Reply:
Since the feature already doesn't work on Main, we will merge this into main to retrieve the partially working implementation and fix it on another branch